### DISCUSSION TITLE
Expression produces a union type that is too complex to represent.ts (2590)

### DISCUSSION THREAD

**Discussion by kritpong-ex** [chakra-ui/chakra-ui#7278] [Q&A] [answered]
Hello, I encountered some problems. And still can't find a way to fix it now.

I'm writing a new project with **ViteJS + TypeScript** and using **Chrakra UI** **Components** and got an error saying **"Expression produces a union type that is too complex to represent.ts (2590)"** which I'm trying to figure out why. but still don't understand Because even if this error occurs But my project can still run. and displayed normally.

**Package details :**
```
"@chakra-ui/react": "^2.4.9"
"@emotion/react": "^11.10.5"
"@emotion/styled": "^11.10.5"
"framer-motion": "^6.5.1
"react": "^18.2.0"
"react-dom": "^18.2.0"
"@types/react": "^18.0.27"
"@types/react-dom": "^18.0.10"
"@vitejs/plugin-react-swc": "^3.0.1"
"typescript": "^4.9.4"
"vite": "^4.0.4"
```

An example of the error I encountered.

![Example 1](https://user-images.githubusercontent.com/22631690/214495125-d77b487b-4c24-491a-8f30-40c7e8f4361b.PNG)

Let's create a new .tsx file and import the Container component.

![Example 2](https://user-images.githubusercontent.com/22631690/214495146-54791cc6-dcaf-4113-b401-09df72a85de9.PNG)

Please advise me.

**Comment by zgid123** [comment, marked as answer]
Can you provide a repository?

**Reply by kritpong-ex** [reply]
Hi zgid123,

My repository [kritpong2006/webapp-report-vitejs](https://github.com/kritpong2006/webapp-report-vitejs.git).

**Reply by zgid123** [reply]
I think it is because you are using `@babel/eslint-parser`, you should config the babel if you wanna use it. You can try this one for `.eslintrc.json`

```json
{
  "parser": "@typescript-eslint/parser",
  "extends": [
    "airbnb",
    "prettier"
  ],
  "plugins": [
    "prettier"
  ],
  "rules": {
    "prettier/prettier": [
      "error"
    ]
  }
}
```

You should install those packages

```
    "@typescript-eslint/eslint-plugin": "^5.49.0",
    "@typescript-eslint/parser": "^5.49.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-plugin-prettier": "^4.2.1",
```

**Reply by kritpong-ex** [reply]
Hi zgid123,

As a test, I tested installing packages and configuring ESLint as per your suggestion. ESLint seems to have additional errors.
As pictured below :

![Error 1](https://user-images.githubusercontent.com/22631690/214784226-d8ec1b22-7c21-4bb7-8043-af2b92c2f0af.png)

So try to find out more about **"react/function-component-definition"** and try to fix according to this document : [function-component-definition.md](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md).

by adding this code to the **.eslintrc** file :

```
"rules": {
    "react/function-component-definition": [
      {
        "namedComponents": [
          "function-declaration",
          "function-expression",
          "arrow-function",
          "function-declaration",
          "function-expression",
          "arrow-function"
        ],
        "unnamedComponents": [
          "function-expression",
          "arrow-function",
          "function-expression",
          "arrow-function"
        ]
      }
    ]
  }
```

The ESLint extended error is gone, but the error **"Expression produces a union type that is too complex to represent.ts (2590)"** is still present.

So I figured the problem might not be caused by ESLint, so I checked the **"JavaScript and TypeScript Nightly"** extensions and tried installing older versions one at a time and kept checking. Until version **4.9.20221031** I found that the above error disappeared. Everything is back to working normally.

![Error 2](https://user-images.githubusercontent.com/22631690/214787381-f0962a13-3eff-48b7-bdc5-3a52d3feafeb.png)

![Error 3](https://user-images.githubusercontent.com/22631690/214787556-342ca847-b227-49c7-9421-1afc920e4fae.png)

I'm not sure if this is the correct solution. For now it works.
I'm studying and still new to TypeScript, if you have any further suggestions please guide me.

Thank you for the advice.

**Reply by zgid123** [reply]
I see. Actually when I clone your repository, I didn't get the issue like you mentioned but other issues (eslint and babel issue). After installing all the packages and change babel parser to typescript parser, I didn't get any issue

**Reply by kritpong-ex** [reply]
Got it, thanks. 👍🏻 

I will review extensions and uninstall extensions that I don't use.

**Comment by jinsley8** [comment]
I just ran into the same issue in VS Code. I thought it was from updating `typescript`, `@typescript-eslint/parser` or `@typescript-eslint/eslint-plugin`.

If you are using VSCode, check which version of TypeScript it is using by opening a JavaScript or TypeScript file and running the `TypeScript: Select TypeScript version` command from the command palette (shortcut: `F1`).

If there is no Use workspace version option, make sure you have TypeScript in your dependencies in package.json and that you have installed your dependencies. If you still do not have it, add `"typescript.enablePromptUseWorkspaceTsdk": true` to `.vscode/settings.json`

This fixed the issue for me.
